gpu/README.md

```diff
@@ -0,0 +1,25 @@
+## Polynomial interpolation methods comparison
+
+Three methods of polynomial interpolation were benchmarked, with different input sizes each time:
+
+- **CPU Lagrange:** Finding the Lagrange polynomial of a set of random points via a naive algorithm (see `math/src/polynomial.rs:interpolate()`)
+- **CPU FFT:** Finding the lowest degree polynomial that interpolates pairs of twiddle factors and Fourier coefficients (the results of applying the Fourier transform to the coefficients of a polynomial) (see `math/src/polynomial.rs:interpolate_fft()`).
+- **GPU FFT (Metal):** Same as CPU FFT but the FFT algorithm is run on the GPU via the Metal API (Apple).
+
+these were run with criterion-rs in a MacBook Pro M1 (18.3), statistically measuring the total run time of one iteration. The field used was a 256 bit STARK-friendly prime field.
+
+All values of time are in milliseconds. Those cases which were greater than 30 seconds were marked respectively as they're too slow and weren't worth to be benchmarked. The input size refers to *d + 1* where *d* is the polynomial's degree (so size is amount of coefficients).
+
+| Input size | CPU Lagrange | CPU FFT   | GPU FFT (Metal) |
+|------------|--------------|-----------|-----------------|
+| 2^4        | 2.1973 ms    | 0.2104 ms | 2.5050 ms       |
```

Comment:
nit, I would use 1 precision number, i.e 2.5 ms instead of 2.5050 ms

Reply:
Done!